src/Microsoft.AspNetCore.Sockets.Client/ServerSentEventsTransport.cs

```diff
@@ -0,0 +1,231 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipelines;
+using System.IO.Pipelines.Text.Primitives;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Formatting;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Sockets.Internal.Formatters;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public class ServerSentEventsTransport : ITransport
+    {
+        private static readonly string DefaultUserAgent = "Microsoft.AspNetCore.SignalR.Client/0.0.0";
+        private static readonly ProductInfoHeaderValue DefaultUserAgentHeader = ProductInfoHeaderValue.Parse(DefaultUserAgent);
+
+        private readonly HttpClient _httpClient;
+        private readonly ILogger _logger;
+        private readonly CancellationTokenSource _transportCts = new CancellationTokenSource();
+        private readonly ServerSentEventsMessageParser _parser = new ServerSentEventsMessageParser();
+
+        private IChannelConnection<SendMessage, Message> _application;
+
+        public Task Running { get; private set; } = Task.CompletedTask;
+
+        public ServerSentEventsTransport(HttpClient httpClient)
+            : this(httpClient, null)
+        { }
+
+        public ServerSentEventsTransport(HttpClient httpClient, ILoggerFactory loggerFactory)
+        {
+            if (httpClient == null)
+            {
+                throw new ArgumentNullException(nameof(_httpClient));
+            }
+
+            _httpClient = httpClient;
+            _logger = (loggerFactory ?? NullLoggerFactory.Instance).CreateLogger<ServerSentEventsTransport>();
+        }
+
+        public Task StartAsync(Uri url, IChannelConnection<SendMessage, Message> application)
+        {
+            _logger.LogInformation("Starting {0}", nameof(ServerSentEventsTransport));
```

Comment:
Use `"Starting {transportName}"`. In Logging (and only in logging) you can use names instead of indexes. The order the tokens appear has to match the input array, and the names and values are stored in key-value pairs for loggers that support it.